lib/security/sha256.js

```diff
@@ -0,0 +1,21 @@
+var CryptoJS = require("crypto-js");
+var q = require('q');
+
+/**
+* Generate a SHA256 hash value for some given text.
+* @param text - the text to generate a hash value for,
+*/
+function hash(text) {
+  var deferred = q.defer();
```

Comment:
Is this function exposed publicly or is it internal? If it's public, I would recommend keeping it async for future proofing, in case you ever decide to use a different encryption library that is async.